File: tests/fixtures/real_world/ruby/taint/yaml_deser.rb
require 'sinatra'
require 'yaml'

post '/parse' do
  data = request.body.read
  config = YAML.load(data)
  config.to_s
end

post '/parse-safe' do
  data = request.body.read
  config = YAML.safe_load(data)
  config.to_s
end

File: tests/fixtures/real_world/python/taint/yaml_deser.py
from flask import Flask, request
import yaml

app = Flask(__name__)

@app.route('/parse', methods=['POST'])
def parse_config():
    data = request.get_data()
    config = yaml.load(data, Loader=yaml.FullLoader)
    return str(config)

@app.route('/parse-safe', methods=['POST'])
def parse_config_safe():
    data = request.get_data()
    config = yaml.safe_load(data)
    return str(config)
